hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -582,7 +582,7 @@ object DataSourceWriteOptions {
 
   val ENABLE_MERGE_INTO_PARTIAL_UPDATES: ConfigProperty[Boolean] = ConfigProperty
     .key("hoodie.spark.sql.merge.into.partial.updates")
-    .defaultValue(true)
+    .defaultValue(false)
```

Comment:
why disable the partial update, is there any known bug?

Reply:
I think @yihua  and @nsivabalan want to use the config to control the compaction code path based on partial updates. Therefore, logically the default value should false; otherwise, the flag is meaningless.